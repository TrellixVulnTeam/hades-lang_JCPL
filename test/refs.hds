
def main(): Void {
    val b = true

    print_bool(ref b)
}

extern def puts(*Byte): Void = puts

def print_bool(value: ref Bool): Void {
    if *value {
        puts(b"true")
    } else {
        puts(b"false")
    }
}