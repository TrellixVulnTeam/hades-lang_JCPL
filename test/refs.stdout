true
message
message
message
message
Updated message
Updated message again
Updated through ref